lib/rubocop/cop/style/multiline_list.rb

```diff
@@ -0,0 +1,184 @@
+      #
+      # * The first parameter in a method definition.
+      # * The first argument in a method call.
+      # * The first element in an array.
+      # * The first element in a hash.
+      #
+      # when the list of parameters, arguments or elements spans multiple lines
+      # It then ensures that the first such element does not also occur on the
+      # object's first line.
+      #
+      # It supports three configuration parameters:
+      #
+      # * EnableMethodParams - boolean
+      # * EnableMethodArgs - boolean
+      # * EnableHashElements - boolean
+      # * EnableArrayElements - boolean
+      #
+      # @example
+      #
+      #     # bad with EnableMethodParams
+      #     def method(foo, bar,
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # good with EnableMethodParams
+      #     def method(
+      #         foo, bar,
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # ignored with EnableMethodParams
+      #     def method foo,
+      #         bar
+      #       do_something
+      #     end
+      #
+      #     # bad with EnableMethodArgs
+      #     method(foo, bar,
+      #       baz)
+      #
+      #     # good with EnableMeethodArgs
+      #     method(
+      #       foo, bar,
+      #       baz)
+      #
+      #     # ignored with EnableMethodArgs
+      #     method foo, bar,
+      #       baz
+      #
+      #     # bad with EnableHashElements
+      #     { a: 1,
+      #       b: 2}
+      #
+      #     # good with EnableHashElements
+      #     {
+      #       a: 1,
+      #       b: 2 }
+      #
+      #     # bad with EnableArrayElements
+      #     [ :a,
+      #       :b]
+      #
+      #     # good with EnableArrayElements
+      #     [
+      #       :a,
+      #       :b]
+      #
+      class MultilineList < Cop
+        MSG = 'Move all elements below the first line of a multi-line list.'
+
+        def on_def(node)
+          return unless cop_config['EnableMethodParams']
+
+          _name, args, _body = *node
+          args = args.to_a
+
+          check_method(node, args)
+        end
+
+        def on_send(node)
+          return unless cop_config['EnableMethodArgs']
+
+          _receiver, _name, *args = *node
+
+          check_method(node, args)
+        end
+
+        def on_array(node)
+          return unless cop_config['EnableArrayElements']
+
+          # Check if we are using an implicit array.
+          source = node.loc.expression.source_line[node.loc.column..-1].lstrip
+
+          return if !explicit_array?(source) && !assignment?(node.parent, node)
+
+          check_children(node, node.children)
+        end
+
+        def on_hash(node)
+          return unless cop_config['EnableHashElements']
+
+          # Check if we are using an implicit hash.
+          source = node.loc.expression.source_line[node.loc.column..-1]
+
+          if source =~ /^\s*\{/
```

Comment:
We usually avoid regexp matching when there's a alternative such as `node.loc.begin` for detecting braces in this case.